File: Factory Pattern/Cpp/PizzaStore/PizzaStore.cpp
#pragma once

#include <string>

#include "../Pizza/Pizza.cpp"

class PizzaStore {
protected:
    virtual Pizza* createPizza(string type) = 0;

public:
    Pizza* orderPizza(string type) {
        Pizza* pizza;

        pizza = createPizza(type);

        pizza->prepare();
        pizza->bake();
        pizza->cut();
        pizza->box();

        return pizza;
    }
};

File: Patterns/Factory Pattern/CSharp/PizzaStore/PizzaStore.cs
using Pizza;

namespace PizzaStore
{
    public abstract class PizzaStore
    {
        protected abstract Pizza.Pizza? CreatePizza(string type);

        public Pizza.Pizza? OrderPizza(string type) {
            Pizza.Pizza? pizza = CreatePizza(type);

            pizza?.Prepare();
            pizza?.Bake();
            pizza?.Cut();
            pizza?.Box();

            return pizza;
        }
    }
}